tests/main/ubuntu-core-netplan-apply/task.yaml

```diff
@@ -0,0 +1,39 @@
+summary: Ensure that netplan apply works on Ubuntu Core with network-setup-control
+
+details: |
+    Netplan apply is used to apply network configuration to the system
+
+environment:
+    NETPLAN: io.netplan.Netplan
+
+# only run on ubuntu core systems
+systems:
+    # TODO: enable on uc18 when core18 is rebuilt with a newer netplan.io pkg
+    - ubuntu-core-16-*
+
+prepare: |
+    #shellcheck source=tests/lib/snaps.sh
+    . "$TESTSLIB"/snaps.sh
+    snap install test-snapd-netplan-apply --edge
+
+restore: |
+    echo "nothing to restore"
+
+execute: |
+    echo "The interface is disconnected by default"
+    snap connections test-snapd-netplan-apply | MATCH 'network-setup-control +test-snapd-netplan-apply:network-setup-control +- +-'
+
+    echo "Running netplan apply without network-setup-control fails"
+    if test-snapd-netplan-apply.netplan apply; then
+        echo "Expected access denied error for netplan apply"
+        exit 1
+    fi
+
+    echo "When the interface is connected"
+    snap connect test-snapd-netplan-apply:network-setup-control
+
+    echo "Running netplan apply now works"
+    if ! test-snapd-netplan-apply.netplan apply; then
+        echo "Unexpected error running netplan apply"
+        exit 1
+    fi
```

Comment:
Can we somehow check that netplan did actually run here? Does it create some log or state or something that we might check if it got an updated timestamp or similar?

Reply:
Yes, we could check that the Network service was stopped and started again. I'll add this

Reply:
It's fine to do this in a followup if you prefer it this way

Reply:
No, sorry I was just busy with other things the PR is now updated to do this.